int32 deta_x
int32 deta_y
